internal/provider/log_export_config_resource.go

```diff
@@ -311,18 +314,21 @@ func loadLogExportIntoTerraformState(
 		apiRedact = types.BoolValue(spec.GetRedact())
 	}
 
-	var apiRegion types.String
-	if spec.Region == nil {
-		apiRegion = types.StringNull()
-	} else {
-		apiRegion = types.StringValue(spec.GetRegion())
+	apiRegion := types.StringNull()
+	if spec.Region != nil && *spec.Region != "" {
```

Comment:
We need to check for both nil and empty string. This mainly what caused the bug mentioned in the commit message